sentinelhub/data_utils.py

```diff
@@ -0,0 +1,154 @@
+    """Transform statistics into pandas.DataFrame entry
+
+    :param interval_output: An input representation of statistics of an aggregation interval.
+    :param exclude_stats: Statistics that will be excluded from output.
+    :return: Statistics as a pandas.DataFrame entry.
+    """
+    stat_entry: Dict[str, Union[List[float], float]] = {}
+    for output_name, output_data in interval_output.items():
+        for band_name, band_values in output_data["bands"].items():
+            band_stats = band_values["stats"]
+            # statistics are not valid when sample count equals to no data count
+            if band_stats["sampleCount"] == band_stats["noDataCount"]:
+                break
+
+            for stat_name, value in band_stats.items():
+                if stat_name not in exclude_stats:
+                    col_name = f"{output_name}_{band_name}_{stat_name}"
+                    if stat_name == "percentiles":
+                        stat_entry.update(
+                            {
+                                f"{col_name}_{percentile_name}": percentile_value
+                                for percentile_name, percentile_value in value.items()
+                            }
+                        )
+                    else:
+                        stat_entry[col_name] = value
+
+                if "histogram" in band_values:
+                    band_bins = band_values["histogram"]["bins"]
+                    hist_bins_value, hist_counts_value = _extract_hist(band_bins)
+                    stat_entry.update(
+                        {
+                            f"{output_name}_{band_name}_bins": hist_bins_value,
+                            f"{output_name}_{band_name}_counts": hist_counts_value,
+                        }
+                    )
+
+    return stat_entry
+
+
+def _extract_response_data(response_data: List[JsonDict], exclude_stats: List[str]) -> List[Dict[str, Any]]:
+    """Transform Statistical API response into a pandas.DataFrame
+
+    :param response_data: An input representation of Statistical API response. The response is a list of JsonDict and
+        each contains the statistics of an aggregation interval.
+    :param exclude_stats: Statistics that will be excluded from output.
+    :return: DataFrame entries of all aggregation intervals of a single geometry.
+    """
+    df_entries = []
+    for interval in response_data:
+        if "outputs" in interval:
+            df_entry: Dict[str, Any] = _extract_stats(interval["outputs"], exclude_stats)
+            if df_entry:
+                df_entry["interval_from"] = parse_time(interval["interval"]["from"])
+                df_entry["interval_to"] = parse_time(interval["interval"]["to"])
+                df_entries.append(df_entry)
+
+    return df_entries
+
+
+def statistical_to_dataframe(result_data: List[JsonDict], exclude_stats: Optional[List[str]] = None) -> Any:
```

Comment:
Rethink the need of `include_histograms`. The histogram data is in the result only if it is requested in the payload. Doesn't make sense to skip it when the data is there.